tests/VocabularyManager.UnitTests/Services/VocabularyStoreManagerTest.cs

```diff
@@ -0,0 +1,107 @@
+﻿using Ardalis.Specification;
+using NSubstitute;
+using VocabularyManager.Core.Entities;
+using VocabularyManager.Core.Interfaces;
+using VocabularyManager.Core.Services;
+using VocabularyManager.Core.Specifications;
+
+namespace VocabularyManager.UnitTests.Services
+{
+    public class VocabularyStoreManagerTest
+    {
+        [Fact]
+        public async Task Add_Word_To_Word_List_Test1()
+        {
+            //Arrange
+            Vocabulary vocabulary = new Vocabulary("some word list", null);
+            Word word = new Word("word", "noun", "a1", null);
+            int vocabularyId = 0;
+            IRepositoryBase<Vocabulary> repository = Substitute.For<IRepositoryBase<Vocabulary>>();
+            repository
+                .GetByIdAsync(vocabularyId)
+                .Returns(
+                    Task.Run(
+                        () => new Vocabulary(vocabulary.Name, vocabulary.SourceUrl)
+                        { Id = vocabularyId, Words = new List<Word>() })
+            ); ;
+            IVocabularyStoreManager vocabularyStoreService = new VocabularyStoreManager(repository);
+
+            //Act
+            int? wordId = await vocabularyStoreService.AddWord(word, vocabularyId);
+
+            //Assert
+            Assert.IsType<int>(wordId);
+            Assert.NotNull(wordId);
+        }
+        [Fact]
+        public async Task Create_Word_List_Test1()
+        {
+            // Arrange
+            Vocabulary vocabulary = new Vocabulary("some word list", null);
+            int expectedId = 0;
```

Comment:
feed input data with some generator like this
https://github.com/AutoFixture/AutoFixture